pkg/resourcemanager/rediscaches/redis/rediscache_reconcile.go

```diff
@@ -90,18 +87,15 @@ func (rc *AzureRedisCacheManager) Ensure(ctx context.Context, obj runtime.Object
 
 		// handle the error
 		if helpers.ContainsString(catchInProgress, azerr.Type) {
-			instance.Status.Message = "RedisCache exists but may not be ready"
+			instance.Status.SetProvisioning("RedisCache exists but may not be ready")
 			return false, nil
 		} else if helpers.ContainsString(catchKnownError, azerr.Type) {
+			// TODO: Not sure why we want this to be provisioning=false?
```

Comment:
It comes down to a question of what we want `provisioning` to mean... I think what the original ASO authors were trying to convey with it was: "I am currently waiting for Azure to actually make the thing", and so they didn't count "waiting in Kubernetes for some reason" as `provisioning = true`.

I don't think that's really a good interpretation though because it's not stable: We set `provisioning = false` here but also return `false` which means "try again" - causing the reconcile loop to come back through and it the `SetProvisioning` on line 71... so the resource is going to blip between provisioning/not until whatever `knownError` is resolved. I think it's clearer to just say that it's provisioning the whole time and also include the message if needed explaining what is going on.

Since we already set the message to include the error details on line 74 I think you can just delete this line.